pointcloud/basic_octree.go

```diff
@@ -50,21 +51,24 @@ type basicOctreeNode struct {
 	maxVal   int
 }
 
-// NewBasicOctree creates a new basic octree with specified center, side and metadata.
-func NewBasicOctree(center r3.Vector, sideLength float64) (*BasicOctree, error) {
+// NewBasicOctree creates a new basic octree with specified center, side and confidenceThreshold.
+// if the confidenceThreshold is out of the allowable limits the default will be used.
+func NewBasicOctree(center r3.Vector, sideLength float64, confidenceThreshold int) (*BasicOctree, error) {
 	if sideLength <= 0 {
 		return nil, errors.Errorf("invalid side length (%.2f) for octree", sideLength)
 	}
-
-	octree := &BasicOctree{
-		node:       newLeafNodeEmpty(),
-		center:     center,
-		sideLength: sideLength,
-		size:       0,
-		meta:       NewMetaData(),
+	if confidenceThreshold < 0 || confidenceThreshold > 100 {
+		confidenceThreshold = defaultConfidenceThreshold
```

Comment:
Passing an invalid parameter should be an error, not a silent update to a different default

Reply:
Honestly theres no reason to even care about erroring here we should let it be whatever